TG-14

evelynda1985 opened the pull request:
Se modificaron lsa clases Ingrediente, Solido y Liquido para que el metodo usarIngrdiente reciba como paramentros ingredientes de tipo Ingrediente.

Tambien se creo la clase Chef en el paque receta.Controller y se creo el metodo CocinarReceta que compara los ingredientes de la receta con los ingredientes de la Cocina.

Problema: no puedo hacer la prueba en el main, me saca error iEmpanada1.usarIngrediente(empanada.getIngrediente());

ideas?

Commit:
TG-6  TG-12
Se crearon la interface Cargador que hace el llamado a los metodos cargarCocina y cargarReceta. 
Se creo CargaManual que carga las Cocinas y Recetas que estaban en el Main.
El main se simplifico al llamado de las cocinas y recetas por medio de la interface.

Commit:
TG-6  TG-12
Se modifico el main, para que imprimiera cada cocina, segun la posicion de i.

Commit:
TG-13  TG-14
Se modificaron lsa clases Ingrediente, Solido y Liquido para que el metodo usarIngrdiente reciba como paramentros ingredientes de tipo Ingrediente.

Tambien se creo la clase Chef en el paque receta.Controller y se creo el metodo CocinarReceta que compara los ingredientes de la receta con los ingredientes de la Cocina.

Problema: no puedo hacer la prueba en el main, me saca error iEmpanada1.usarIngrediente(empanada.getIngrediente());

ideas?

jramirezl commented on src/receta/controller/Chef.java:
```diff
@@ -0,0 +1,31 @@
+        
+        for (int i = 0; i < ingredientesReceta.length; i++){
+            for (int j = 0; j < ingredientesCocina.length; j++){
+                if (ingredientesReceta[i].equals(ingredientesCocina[j]))
```
para que esto se cumpla se debe reimplementar el metodo equals en la clase abstracta ingredientes. Hay otra opcion mas manual

Commit:
TG-13  TG-14
Se implemento el  metodo equals en la clase Ingrediente.

jramirezl commented on src/receta/dto/def/Ingrediente.java:
```diff
@@ -66,7 +66,16 @@ public void setMedida(String medida) {
         return "Nombre: " + nombre + " Cantidad: " +cantidad + " Medida: " + medida; 
     }
+    
+    public boolean equals(Ingrediente ingrediente){
```
esta bien pero debe contener @Override y ademas implementar el metodo hashCode
Revisa el ejemplo de 2. Classic Way

Commit:
TG-13  TG-14
se compara el nombre en la iteracion de cocinarReceta y se elimina el metodo equals en la clase Ingrediente. :p

jramirezl commented on src/receta/controller/Chef.java:
```diff
@@ -0,0 +1,31 @@
+        
+        for (int i = 0; i < ingredientesReceta.length; i++){
+            for (int j = 0; j < ingredientesCocina.length; j++){
+                if (ingredientesReceta[i].getNombre().equals(ingredientesCocina[j].getNombre()))
```
Bien!!!!

jramirezl reviewed (approved):
Se ve bn
